delivery_free_fee_removal/tests/test_delivery_free_fee_removal.py

```diff
@@ -58,3 +80,25 @@ def test_delivery_free_fee_removal_free_fee(self):
         self.assertRecordValues(
             delivery_line, [{"is_free_delivery": True, "qty_to_invoice": 0}]
         )
+        self.deliver_and_invoice_order(self.sale)
+
+    def deliver_and_invoice_order(self, order):
+        """Check delivery line state for invoicing and delivery."""
+        # Deliver the order
+        picking = order.picking_ids
+        self.assertEqual(picking.state, "assigned")
+        picking.move_line_ids[0].qty_done = 1.0
+        picking.action_done()
+        self.assertEqual(picking.state, "done")
+        # Invoice the order
+        self.wizard_obj = self.env["sale.advance.payment.inv"]
+        wizard = self.wizard_obj.with_context(
+            active_ids=order.ids, active_model=order._name
+        ).create({})
+        wizard.create_invoices()
+        self.assertTrue(order.invoice_ids)
+        # Check delivery lines are properly invoiced and delivered
+        fee_line = order.order_line.filtered("is_delivery")
+        self.assertEqual(fee_line.invoice_status, "invoiced")
+        # Check the order is fully invoiced
+        self.assertEqual(self.sale.invoice_status, "invoiced")
```

Comment:
My reading of the code tells me that the fee line is supposed to have `invoice_status` equal to `invoiced` even before the invoice is created, as soon as we confirm the SO. If so, only testing this status in the `assertRecordValues` of the tests would have been enough:

```
        self.assertRecordValues(
            delivery_line, [{"is_free_delivery": False, "qty_to_invoice": 1, "invoice_status": "invoiced"}]
        )
```

(The `invoice_status` of the sale being equal to "invoiced" when all the lines are "invoiced" is a property of the "sale" module, which should be tested (and maybe is) there.)

Reply:
Did that, thanks for your review